server/src/gnome/prompter.rs

```diff
@@ -0,0 +1,287 @@
+// org.gnome.keyring.Prompter
+// https://gitlab.gnome.org/GNOME/gcr/-/blob/main/gcr/org.gnome.keyring.Prompter.xml
+
+use clap::error::Result;
+use oo7::dbus::ServiceError;
+use serde::{Deserialize, Serialize};
+use zbus::{
+    interface, proxy,
+    zvariant::{DeserializeDict, OwnedObjectPath, OwnedValue, SerializeDict, Type, Value},
+};
+
+use crate::{
+    gnome::secret_exchange::SecretExchange,
+    prompt::{Prompt, PromptRole},
+    service::Service,
+};
+
+#[derive(Debug, DeserializeDict, SerializeDict, Type)]
+#[zvariant(signature = "a{sv}")]
+// System prompt properties
+pub struct Properties {
```

Comment:
With this approach, during the second PromptReady call return I'm getting the following error, `Unexpected non-0 padding byte 1`. I can't figure out why. I think this is related to [this](https://gitlab.freedesktop.org/dbus/zbus/-/issues/82). Then again all the fields here are properly renamed using `#[zvariant(rename = "")`

Reply:
Check what gets send over the bus with busctl and see what is going on? Probably a mismatched type

Reply:
Will do, thanks.

Reply:
I tried everything I can think of. I think this is happening due to a bug in zbus. Probably related to zbus issues #1193 and #320.

Reply:
Can you write a minimal reproducer of the issue? I am using the exact same thing in ashpd and it works fine.

Reply:
Will do.

> I am using the exact same thing in ashpd and it works fine.

Yeah, I did see that.

Reply:
Actually the issues you mentioned are about the OwnedValue/Value derive macros which you don't use here. So the issue is very likely in the types you are using inside Properties struct. Maybe they are all passed as strings?

Reply:
So [these](https://gitlab.gnome.org/GNOME/gcr/-/blob/main/gcr/gcr-prompt.c#L95) are the GcrPrompt properties.

Reply:
Show me the dbus traffic instead.

Reply:
https://upload.disroot.org/r/uCEPb8Pu#Z6OOnI1W4Fx5uALHyb8s5IIVaaDAngKl5tPm5m0tk04=

Reply:
Can you confirm this is caused by the properties and not the the reply param of PromptReady?

Reply:
Then keep the Properties struct there and add a todo comment about it so we can investigate it further.

Reply:
got it, thanks! I think I'll have to keep this diff as well. I can't test anything without it :)

Reply:
Just locally.